app/src/ui/rebase/rebase-flow.tsx

```diff
@@ -82,61 +81,81 @@ export class RebaseFlow extends React.Component<IRebaseFlowProps> {
+  private onContinueRebase = async () => {
+    const { dispatcher, repository, workingDirectory, step } = this.props
+    if (step.kind !== RebaseStep.ShowConflicts) {
+      // This shouldn't happen, but needed the type checking.
+      log.error(
+        '[Rebase] Invoked continue of rebase without being in a conflict step.'
+      )
+      this.onFlowEnded()
+      return
+    }
     const { conflictState } = step
 
     const continueRebaseAction = () => {
-      return this.props.dispatcher.continueRebase(
-        this.props.repository,
-        this.props.workingDirectory,
+      return dispatcher.continueRebase(
+        repository,
+        workingDirectory,
         conflictState
       )
     }
 
-    this.props.dispatcher.setRebaseFlowStep(this.props.repository, {
+    return dispatcher.setRebaseFlowStep(repository, {
       kind: RebaseStep.ShowProgress,
       rebaseAction: continueRebaseAction,
     })
   }
 
-  private showRebaseConflictsBanner = (step: ShowConflictsStep) => {
-    this.props.dispatcher.setRebaseFlowStep(this.props.repository, {
+  private onConflictsDialogDismissed = () => {
+    const { dispatcher, repository, step } = this.props
+    if (step.kind !== RebaseStep.ShowConflicts) {
+      // This shouldn't happen, but needed the type checking.
+      log.error(
+        '[Rebase] Cannot show rebase conflict banner without being in a conflict step.'
+      )
+      this.onFlowEnded()
+      return
+    }
+
+    dispatcher.setRebaseFlowStep(repository, {
       kind: RebaseStep.HideConflicts,
     })
 
     const { targetBranch } = step.conflictState
 
-    this.props.onShowRebaseConflictsBanner(this.props.repository, targetBranch)
+    this.props.onShowRebaseConflictsBanner(repository, targetBranch)
   }
 
-  private onConfirmAbortRebase = (step: ShowConflictsStep) => {
-    const { workingDirectory, userHasResolvedConflicts } = this.props
-    const { conflictState } = step
-    const { manualResolutions } = conflictState
-
-    if (userHasResolvedConflicts) {
-      // a previous commit was resolved by the user
-      this.props.dispatcher.setRebaseFlowStep(this.props.repository, {
-        kind: RebaseStep.ConfirmAbort,
-        conflictState,
-      })
+  private onConfirmAbortRebase = async () => {
+    const { workingDirectory, userHasResolvedConflicts, step } = this.props
+    if (step.kind !== RebaseStep.ShowConflicts) {
+      // This shouldn't happen, but needed the type checking.
+      log.error(
+        '[Rebase] Invoked abort of rebase without being in a conflict step.'
+      )
+      this.onFlowEnded()
       return
     }
 
-    // otherwise check the current commit for resolved changes
+    const { conflictState } = step
+    const { manualResolutions } = conflictState
+
     const resolvedConflicts = getResolvedFiles(
       workingDirectory,
       manualResolutions
     )
 
-    if (resolvedConflicts.length > 0) {
+    if (userHasResolvedConflicts || resolvedConflicts.length > 0) {
```

Comment:
This method had two checks one for boolean and one for resolvedConflicts that both led to same popup. I just combined them as I don't think the resolved conflict check is that expensive to warrant the redundant code (looks like just reading the variables passed in).